transport.go

```diff
@@ -94,6 +94,38 @@ func getRequestBodyFromEvent(event *Event) []byte {
 	return nil
 }
 
+func encodeAttachment(enc *json.Encoder, b io.Writer, attachment *Attachment) error {
```

Comment:
Please add a test in `transport_test.go` that tests this function.

Reply:
PTAL at https://github.com/getsentry/sentry-go/pull/670/commits/52caef25b90d3055ea9ee5066536c2493a4d4fed.

Not sure if testing `encodeAttachment()` makes sense in the presence of tests like `TestEnvelopeFromErrorBody`, which already will call that function under the hood if we set `event.attachments` to non-`nil`.

Reply:
Yep, what you added is actually what I had in mind 👍